File: prototype/ts/NormalPoint.ts
import { ExtendedModel, model, prop } from 'mobx-keystone';

import Onto from './Onto';
import Point from './Point';
import Normal from './Normal';
import { NormalPoint as INormalPoint } from './types';


@model('onto/NormalPoint')
class NormalPoint extends ExtendedModel(Onto, {
    Point: prop<Point>(() => new Point({})),
    Normal: prop<Normal>(() => new Normal({})),
}) implements INormalPoint {};

export default NormalPoint;

File: packages/csharp/onto/OnToSharp/src/Types/NormalPoint.cs
using GraphQLCodeGen;
using Stratosphere;


namespace Types {

    public class NormalPoint : OntoNode, Schema.NormalPoint {
        public Normal Normal { get; set; }
        public Point Point { get; set; }

        public NormalPoint(Normal? normal, Point? point) {
            this.Normal = normal ?? new Normal();
            this.Point = point ?? new Point();
        }
    }

}
